pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -858,105 +858,146 @@ func TestPriorityQueue_Pop(t *testing.T) {
-	if err != nil || podGotFromBackoffQ != medPriorityPodInfo.Pod {
-		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPodInfo.Pod.Name, podGotFromBackoffQ.Name)
-	}
+	notInActiveQ := "NotInActive"
 
-	// To simulate the pod is failed in scheduling in the real world, Pop() the pod from activeQ before testing AddUnschedulableIfNotPresent.
-	q.activeQ.Add(podInfo)
-	if p, err := q.Pop(); err != nil || p.Pod != medPriorityPodInfo.Pod {
-		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPodInfo.Pod.Name, p.Pod.Name)
-	}
-	err = q.AddUnschedulableIfNotPresent(logger, q.newQueuedPodInfo(medPriorityPodInfo.Pod, "plugin"), q.SchedulingCycle())
-	if err != nil {
-		t.Fatalf("unexpected error from AddUnschedulableIfNotPresent: %v", err)
-	}
-	if len(q.unschedulablePods.podInfoMap) != 1 {
-		t.Error("Expected unschedulablePods to be 1.")
-	}
-	updatedPod := medPriorityPodInfo.Pod.DeepCopy()
-	updatedPod.Annotations["foo"] = "test"
-	q.Update(logger, medPriorityPodInfo.Pod, updatedPod)
-	rawPodInfo, err = q.podBackoffQ.Pop()
-	podGotFromBackoffQ = rawPodInfo.(*framework.QueuedPodInfo).Pod
-	if err != nil || podGotFromBackoffQ != updatedPod {
-		t.Errorf("Expected: %v after Pop, but got: %v", updatedPod.Name, podGotFromBackoffQ.Name)
-	}
+	tests := []struct {
+		name               string
+		expectedQ          string
+		checkNominatedPods bool
+		// prepareFunc is the function called before the test calls Update(). After prepareFunc,
+		// each test will call Update() with oldPod and newPod returned from it.
+		prepareFunc func(q *PriorityQueue) (oldPod, newPod *v1.Pod)
+	}{
+		{
+			name:      "add highPriorityPodInfo to activeQ",
+			expectedQ: activeQ,
+			prepareFunc: func(q *PriorityQueue) (oldPod, newPod *v1.Pod) {
+				return nil, highPriorityPodInfo.Pod
+			},
+		},
+		{
+			name:      "Update pod that oldPod doesn't exist in the queue",
+			expectedQ: activeQ,
+			prepareFunc: func(q *PriorityQueue) (oldPod, newPod *v1.Pod) {
+				updatedPod := medPriorityPodInfo.Pod.DeepCopy()
+				updatedPod.Annotations["foo"] = "test"
+				return medPriorityPodInfo.Pod, updatedPod
+			},
+		},
+		{
+			name:               "Update highPriorityPodInfo and add a nominatedNodeName to it",
+			expectedQ:          activeQ,
+			checkNominatedPods: true,
+			prepareFunc: func(q *PriorityQueue) (oldPod, newPod *v1.Pod) {
+				return highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod
+			},
+		},
+		{
+			name:      "Updating a pod that is already in activeQ, Pod should remain in activeQ after Update()",
+			expectedQ: activeQ,
+			prepareFunc: func(q *PriorityQueue) (oldPod, newPod *v1.Pod) {
+				// Updating an unschedulable pod which is not in any of the two queues, should
+				// add the pod to activeQ.
+				err := q.Update(logger, unschedulablePodInfo.Pod, unschedulablePodInfo.Pod)
+				if err != nil {
+					t.Errorf("update pod %s error: %v", unschedulablePodInfo.Pod.Name, err)
+				}
```

Comment:
nit:
Let's make it simpler, we check this updating scenario in another test case, so we don't need to try it here. What we have to do is just adding pod to activeQ.

```suggestion
				err := q.Update(logger, nil, unschedulablePodInfo.Pod)
				if err != nil {
					t.Errorf("add pod %s error: %v", unschedulablePodInfo.Pod.Name, err)
				}
```